src/geo/projection/globe_util.js

```diff
@@ -655,6 +655,14 @@ export function isLngLatBehindGlobe(tr: Transform, lngLat: LngLat): boolean {
     return (globeTiltAtLngLat(tr, lngLat) > Math.PI / 2 * 1.01);
 }
 
+export function isLngLatInViewport(tr: Transform, lngLat: LngLat): boolean {
+    const lngLatToPoint = latLngToECEF(lngLat.lat, lngLat.lng);
+    const aabb = Aabb.fromPoints([lngLatToPoint]);
+    const cameraFrustum = tr.getCameraFrustum();
+    const intersectResult = aabb.intersects(cameraFrustum);
```

Comment:
I don't think the point and the frustrum are in the same coordinate space here, since the AABB is in ECEF while the frustrum is in tile space.